providers/component-protocol/protocol/render.go

```diff
@@ -78,6 +78,8 @@ func RunScenarioRender(ctx context.Context, req *cptype.ComponentProtocolRequest
 		}
 
 	} else {
+		// root is always rendered
+		compRending = append(compRending, cptype.RendingItem{Name: req.Protocol.Hierarchy.Root})
```

Comment:
How about event trigger by the root, may be render twice is bad.

Reply:
you are right